File: Go/Algorithms/2351.go
package main

func repeatedCharacter(s string) byte {
	seen := make(map[rune]struct{})
	for _, ch := range s {
		if _, ok := seen[ch]; ok {
			return byte(ch)
		}
		seen[ch] = struct{}{}
	}
	return ' '
}

File: Rust/Algorithms/2351.rs
use std::collections::HashSet;

impl Solution {
    pub fn repeated_character(s: String) -> char {
        let mut seen = HashSet::new();
        for ch in s.chars() {
            if !seen.insert(ch) {
                return ch;
            }
        }
        ' '
    }
}